packages/did-auth-siop-op-authenticator/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -0,0 +1,204 @@
+        }
+      })
+      .catch((error: unknown) => Promise.reject(error))
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.getDidSiopAuthenticationRequestFromRP} */
+  private async getDidSiopAuthenticationRequestFromRP(
+    args: IGetDidSiopAuthenticationRequestFromRpArgs,
+    context: IRequiredContext
+  ): Promise<ParsedAuthenticationRequestURI> {
+    return fetch(`${args.redirectUrl}?stateId=${args.stateId}`)
+      .then(async (response: Response) =>
+        response.status >= 400 ? Promise.reject(new Error(await response.text())) : this.op.parseAuthenticationRequestURI(await response.text())
+      )
+      .catch((error: unknown) => Promise.reject(error))
+  }
+
+  /** {@inheritDoc IDidAuthSiopOpAuthenticator.getDidSiopAuthenticationRequestDetails} */
+  private async getDidSiopAuthenticationRequestDetails(
+    args: getDidSiopAuthenticationRequestDetailsArgs,
+    context: IRequiredContext
+  ): Promise<IAuthRequestDetails> {
+    const presentationDefs = args.verifiedAuthenticationRequest.presentationDefinitions
+    const verifiablePresentations =
+      presentationDefs && presentationDefs.length > 0 ? await this.matchPresentationDefinitions(presentationDefs, args.verifiableCredentials) : []
+    const didResolutionResult = args.verifiedAuthenticationRequest.didResolutionResult
+
+    return {
+      id: didResolutionResult.didDocument!.id,
+      alsoKnownAs: didResolutionResult.didDocument!.alsoKnownAs,
+      vpResponseOpts: verifiablePresentations,
+    }
+  }
+
+  private async matchPresentationDefinitions(
+    presentationDefs: PresentationDefinitionWithLocation[],
+    verifiableCredentials: VerifiableCredential[]
+  ): Promise<IMatchedPresentationDefinition[]> {
+    const presentationExchange = this.getPresentationExchange(verifiableCredentials)
+    return await Promise.all(
+      presentationDefs.map(async (presentationDef: PresentationDefinitionWithLocation) => {
+        const checked = await presentationExchange.selectVerifiableCredentialsForSubmission(presentationDef.definition)
+        if (checked.errors && checked.errors.length > 0) {
+          return Promise.reject(new Error(JSON.stringify(checked.errors)))
+        }
+
+        const matches: SubmissionRequirementMatch[] | undefined = checked.matches
+        if (matches && matches.length == 0) {
+          return Promise.reject(new Error(JSON.stringify(checked.errors)))
+        }
+
+        const verifiablePresentation = await presentationExchange.submissionFrom(presentationDef.definition, verifiableCredentials)
+        return {
+          location: presentationDef.location,
+          format: VerifiablePresentationTypeFormat.LDP_VP,
+          presentation: verifiablePresentation,
+        }
+      })
+    )
+  }
+
+  private getPresentationExchange(verifiableCredentials: VerifiableCredential[]): PresentationExchange {
```

Comment:
This would need to be integrated with Veramo Verifiable Credentials and Storage, but that one can wait for now I guess

Reply:
So I tried using the Veramo credentials from the datastore, but they are incompatible with PE-JS. So we need to do some mapping or start supporting them. 

For now I added a TODO with a link to a ticket